### PR TITLE
[CSL-1747] Use log-warper in the launcher (again)

### PR DESCRIPTION
**Pull request by neongreen**
**Note:** After this is merged, we should also merge this PR to Daedalus: https://github.com/input-output-hk/daedalus/pull/517. It doesn't have to be done immediately (e.g. no builds would break if the latter PR isn't merged, and Daedalus will continue working), but it still should be merged before the next update.

### CODE REVIEW

**Comment by neongreen** [comment]
need to regenerate stack2nix, likely

**Comment by neongreen** [comment]
I tested it on Windows, it launches now but the launcher crashes soon after for some reason. I have another idea that I'm going to try

**Comment by neongreen** [comment]
Okay, really sleep now but I think the [Visual Basic solution](https://superuser.com/questions/62525/run-a-batch-file-in-a-completely-hidden-way) should work, I'll try it tomorrow

File: tools/cardano-sl-tools.cabal [hunk 1]
```diff
@@ -306,9 +306,6 @@ executable cardano-launcher
                        BangPatterns
                        ScopedTypeVariables
 
-  if os(windows)
```
Comment (gromakovsky) [inline tools/cardano-sl-tools.cabal]: Can you justify this change? The commit message doesn't have any clarifications :/

**Comment by neongreen** [comment]
@gromakovsky 

> The commit message doesn't have any clarifications :/

It was just to build an installer. I will create proper commits when I get it working.

**Comment by neongreen** [comment]
@fersel okay, I think this should work! Please test.

**Comment by gromakovsky** [comment]
@neongreen please write which actions should be performed by DevOps in the first message.
Also, don't make the same mistake and don't merge it before you are allowed to do so.

**Comment by fersel** [comment]
Note: "PR to Daedalus" should go to `release\0.8.2` branch.

File: tools/src/launcher/Main.hs [hunk 2]
```diff
@@ -491,3 +546,19 @@ ignoreSIGPIPE = handle (\ex -> case ex of
...
+
+----------------------------------------------------------------------------
+-- SIGKILL
+----------------------------------------------------------------------------
+
+-- | If we're on Linux or macOS, send a SIGKILL to a process.
+maybeTrySIGKILL :: ProcessHandle -> M ()
+maybeTrySIGKILL _h = do
+#ifdef mingw32_HOST_OS
+    logInfo "Not trying to send a SIGKILL because we're on Windows"
```
Comment (deepfire) [inline tools/src/launcher/Main.hs]: Is there really no way to ask a process to terminate on Windows?

It's not very good to have behaviors diverge between OSes, in the long run..

Comment (neongreen) [reply tools/src/launcher/Main.hs]: According to MSDN, `terminateProcess` is already as powerful as it gets

Comment (dcoutts) [reply tools/src/launcher/Main.hs]: The problem on Windows is that there is no standard equivalent of SIGTERM, of asking nicely for a process to terminate. There is only `TerminateProcess()` which is forcible and gives no chance to clean up.

http://www.drdobbs.com/a-safer-alternative-to-terminateprocess/184416547

File: tools/src/launcher/Main.hs [hunk 3]
```diff
@@ -325,7 +326,9 @@ clientScenario logConf node wallet updater nodeTimeout report = do
   where
     killNode nodeHandle nodeAsync = do
         logInfo "Killing the node"
-        liftIO $ Process.terminateProcess nodeHandle
+        liftIO (tryAny (Process.terminateProcess nodeHandle)) >>= \case
```
Comment (dcoutts) [inline tools/src/launcher/Main.hs]: Why do we need to catch any exception? Isn't it only normal `IOException` we need to catch?

**Pull request merged**